v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/BigtableChangeStreamsToPubSub.java

```diff
@@ -0,0 +1,492 @@
+    optionsClass = BigtableChangeStreamsToPubSubOptions.class,
+    optionsOrder = {
+      BigtableChangeStreamsToPubSubOptions.class,
+      BigtableCommonOptions.ReadChangeStreamOptions.class,
+      BigtableCommonOptions.ReadOptions.class
+    },
+    skipOptions = {
+      "bigtableReadAppProfile",
+      "bigtableAdditionalRetryCodes",
+      "bigtableRpcAttemptTimeoutMs",
+      "bigtableRpcTimeoutMs"
+    },
+    flexContainerName = "bigtable-changestreams-to-pubsub",
+    contactInformation = "https://cloud.google.com/support")
+public final class BigtableChangeStreamsToPubSub {
+
+  /** String/String Coder for {@link FailsafeElement}. */
+  public static final FailsafeElementCoder<String, String> FAILSAFE_ELEMENT_CODER =
+      FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of());
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigtableChangeStreamsToPubSub.class);
+
+  private static final String USE_RUNNER_V2_EXPERIMENT = "use_runner_v2";
+
+  /**
+   * Main entry point for executing the pipeline.
+   *
+   * @param args The command-line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    LOG.info("Starting to replicate change records from Cloud Bigtable change streams to PubSub");
+
+    BigtableChangeStreamsToPubSubOptions options =
+        PipelineOptionsFactory.fromArgs(args)
+            .withValidation()
+            .as(BigtableChangeStreamsToPubSubOptions.class);
+
+    run(options);
+  }
+
+  private static void validateOptions(BigtableChangeStreamsToPubSubOptions options) {
+    if (options.getDlqRetryMinutes() <= 0) {
+      throw new IllegalArgumentException("dlqRetryMinutes must be positive.");
+    }
+    if (options.getDlqMaxRetries() < 0) {
+      throw new IllegalArgumentException("dlqMaxRetries cannot be negative.");
+    }
+  }
+
+  private static void setOptions(BigtableChangeStreamsToPubSubOptions options) {
+    options.setStreaming(true);
+    options.setEnableStreamingEngine(true);
+
+    // Add use_runner_v2 to the experiments option, since change streams connector is only supported
+    // on Dataflow runner v2.
+    List<String> experiments = options.getExperiments();
+    if (experiments == null) {
+      experiments = new ArrayList<>();
+    }
+    boolean hasUseRunnerV2 = false;
+    for (String experiment : experiments) {
+      if (experiment.toLowerCase().equals(USE_RUNNER_V2_EXPERIMENT)) {
+        hasUseRunnerV2 = true;
+        break;
+      }
+    }
+    if (!hasUseRunnerV2) {
+      experiments.add(USE_RUNNER_V2_EXPERIMENT);
+    }
+    options.setExperiments(experiments);
+  }
+
+  /**
+   * Runs the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  public static PipelineResult run(BigtableChangeStreamsToPubSubOptions options) {
+    LOG.info(
```

Comment:
missing whitespace between format and encoding. Suggest doing this:

`LOG.info("Requested message format: {}, message encoding: {}", options.getMessageFormat(), options.getMessageEncoding());`